diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 70b62d5b..4ff4724a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -32,7 +32,9 @@ if (NOT WIN32)
     DEPENDS espeak-ng-bin
   )
 endif()
-install(TARGETS espeak-ng-bin)
+if (NOT CMAKE_SYSTEM_NAME STREQUAL "iOS")
+  install(TARGETS espeak-ng-bin)
+endif()
 install(DIRECTORY include/espeak include/espeak-ng TYPE INCLUDE)
 
 if (ESPEAK_COMPAT AND NOT WIN32)
