diff --git a/modules/audio_processing/aec3/matched_filter_avx2.cc b/modules/audio_processing/aec3/matched_filter_avx2.cc
index eacd6486b1..06b7e9faa0 100644
--- a/modules/audio_processing/aec3/matched_filter_avx2.cc
+++ b/modules/audio_processing/aec3/matched_filter_avx2.cc
@@ -82,14 +82,17 @@ void MatchedFilterCore_AccumulatedError_AVX2(size_t x_start_index,
       s_inst_256_8 = _mm256_mul_ps(h_k_8, x_k_8);
       s_inst_hadd_256 = _mm256_hadd_ps(s_inst_256, s_inst_256_8);
       s_inst_hadd_256 = _mm256_hadd_ps(s_inst_hadd_256, s_inst_hadd_256);
-      s_acum += s_inst_hadd_256[0];
-      e_128[0] = s_acum - y[i];
-      s_acum += s_inst_hadd_256[4];
-      e_128[1] = s_acum - y[i];
-      s_acum += s_inst_hadd_256[1];
-      e_128[2] = s_acum - y[i];
-      s_acum += s_inst_hadd_256[5];
-      e_128[3] = s_acum - y[i];
+      const float* s_inst_hadd_256_f = reinterpret_cast<const float*>(&s_inst_hadd_256);
+      alignas(16) float e_128_f[4];
+      s_acum += s_inst_hadd_256_f[0];
+      e_128_f[0] = s_acum - y[i];
+      s_acum += s_inst_hadd_256_f[4];
+      e_128_f[1] = s_acum - y[i];
+      s_acum += s_inst_hadd_256_f[1];
+      e_128_f[2] = s_acum - y[i];
+      s_acum += s_inst_hadd_256_f[5];
+      e_128_f[3] = s_acum - y[i];
+      e_128 = _mm_load_ps(e_128_f);
 
       __m128 acum_error = _mm_loadu_ps(a_p);
       acum_error = _mm_fmadd_ps(e_128, e_128, acum_error);
@@ -210,8 +213,9 @@ void MatchedFilterCore_AVX2(size_t x_start_index,
     x2_sum_256 = _mm256_add_ps(x2_sum_256, x2_sum_256_8);
     s_256 = _mm256_add_ps(s_256, s_256_8);
     __m128 sum = hsum_ab(x2_sum_256, s_256);
-    x2_sum += sum[0];
-    s += sum[1];
+    const float* sum_f = reinterpret_cast<const float*>(&sum);
+    x2_sum += sum_f[0];
+    s += sum_f[1];
 
     // Compute the matched filter error.
     float e = y[i] - s;