diff --git a/src/include/espeak-ng/speak_lib.h b/src/include/espeak-ng/speak_lib.h
index 9ec4cafb..23ee6936 100644
--- a/src/include/espeak-ng/speak_lib.h
+++ b/src/include/espeak-ng/speak_lib.h
@@ -29,7 +29,9 @@
 #include <stddef.h>
 
 #if defined(_WIN32) || defined(_WIN64)
-#ifdef LIBESPEAK_NG_EXPORT
+#if defined(ESPEAK_NG_STATIC)
+#define ESPEAK_API
+#elif defined(LIBESPEAK_NG_EXPORT)
 #define ESPEAK_API __declspec(dllexport)
 #else
 #define ESPEAK_API __declspec(dllimport)