diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 70b62d5b..4ff4724a 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -32,7 +32,9 @@ if (NOT WIN32)
     DEPENDS espeak-ng-bin
   )
 endif()
-install(TARGETS espeak-ng-bin)
+if (NOT CMAKE_SYSTEM_NAME STREQUAL "iOS")
+  install(TARGETS espeak-ng-bin)
+endif()
 install(DIRECTORY include/espeak include/espeak-ng TYPE INCLUDE)
 
 if (ESPEAK_COMPAT AND NOT WIN32)
diff --git a/src/include/espeak-ng/speak_lib.h b/src/include/espeak-ng/speak_lib.h
index 9ec4cafb..23ee6936 100644
--- a/src/include/espeak-ng/speak_lib.h
+++ b/src/include/espeak-ng/speak_lib.h
@@ -29,7 +29,9 @@
 #include <stddef.h>
 
 #if defined(_WIN32) || defined(_WIN64)
-#ifdef LIBESPEAK_NG_EXPORT
+#if defined(ESPEAK_NG_STATIC)
+#define ESPEAK_API
+#elif defined(LIBESPEAK_NG_EXPORT)
 #define ESPEAK_API __declspec(dllexport)
 #else
 #define ESPEAK_API __declspec(dllimport)
