diff --git a/Kokoro.cpp b/Kokoro.cpp
index 28ab261..b93928b 100644
--- a/Kokoro.cpp
+++ b/Kokoro.cpp
@@ -14,7 +14,7 @@ std::vector<float> trim_audio(const std::vector<float>& audio, int sample_rate,
     return audio; 
 }
 
-Kokoro::Kokoro(const std::string& model_path, const std::string& voices_path, const std::string& vocab_path) 
+Kokoro::Kokoro(const std::string& model_path, const std::string& voices_path, const std::string& dict_dir, const std::string& vocab_path) 
     : env_(ORT_LOGGING_LEVEL_WARNING, "Kokoro")
 {
     // Initialize session options
@@ -63,7 +63,9 @@ Kokoro::Kokoro(const std::string& model_path, const std::string& voices_path, co
     }
 
     // Initialize Tokenizer
-    tokenizer_ = std::make_unique<Tokenizer>(TokenizerConfig{}, vocab);
+    TokenizerConfig tok_cfg;
+    tok_cfg.dict_dir = dict_dir;
+    tokenizer_ = std::make_unique<Tokenizer>(tok_cfg, vocab);
 }
 
 Kokoro::~Kokoro() {
