diff --git a/Kokoro.h b/Kokoro.h
index 0887c65..4d7d945 100644
--- a/Kokoro.h
+++ b/Kokoro.h
@@ -20,7 +20,7 @@ const int SAMPLE_RATE = 24000;      // Example value
 
 class Kokoro {
 public:
-    Kokoro(const std::string& model_path, const std::string& voices_path, const std::string& vocab_path = "dict/vocab.txt");
+    Kokoro(const std::string& model_path, const std::string& voices_path, const std::string& dict_dir, const std::string& vocab_path);
     ~Kokoro();